lib/media/ver10/get_profile.ex

```diff
@@ -15,15 +15,18 @@ defmodule Onvif.Media.Ver10.GetProfile do
   end
 
   def response(xml_response_body) do
-    xml_response_body
-    |> parse(namespace_conformant: true, quiet: true)
-    |> xpath(
-      ~x"//s:Envelope/s:Body/tr2:GetProfilesResponse/tr2:Profiles"el
-      |> add_namespace("s", "http://www.w3.org/2003/05/soap-envelope")
-      |> add_namespace("trt", "http://www.onvif.org/ver10/media/wsdl")
-      |> add_namespace("tt", "http://www.onvif.org/ver10/schema")
-    )
-    |> Onvif.Media.Ver10.Profile.parse()
-    |> Onvif.Media.Ver10.Profile.to_struct()
+    res =
+      xml_response_body
+      |> parse(namespace_conformant: true, quiet: true)
+      |> xpath(
+        ~x"//s:Envelope/s:Body/tr2:GetProfilesResponse/tr2:Profiles"el
+        |> add_namespace("s", "http://www.w3.org/2003/05/soap-envelope")
+        |> add_namespace("trt", "http://www.onvif.org/ver10/media/wsdl")
+        |> add_namespace("tt", "http://www.onvif.org/ver10/schema")
+      )
+      |> Onvif.Media.Ver10.Profile.parse()
+      |> Onvif.Media.Ver10.Profile.to_struct()
+
+    {:ok, res}
```

Comment:
`to_struct/1` uses `apply_action/2` which returns an `:ok` tuple already, so you are double wrapping this one (and the next file does this too)